src/Tokenizer.js

```diff
@@ -479,7 +495,16 @@ module.exports = class Tokenizer {
       } else {
         title = cap[3] ? cap[3].slice(1, -1) : '';
       }
-      href = href.trim().replace(/^<([\s\S]*)>$/, '$1');
+
+      href = href.trim();
+      if (href.startsWith('<')) {
+        if (this.options.pedantic && !trimmedUrl.endsWith('>')) {
```

Comment:
There are multiple checks for the pedantic option. Is there some way all of the pedantic logic can be handled together?

Reply:
That would require duplicating a lot of the logic since most of the logic is the same.

Reply:
See my PR on your branch.  Broke out my old boolean logic notes and reordered some things around. Not sure if it's "better", but it's more legible to me.